Automattic/sensei
# Add new context notices class and submitted quiz notice

Fixes #4407

### Changes proposed in this Pull Request

* It introduces the new `Sensei_Context_Notices` class to handle notices. It's a more independent class than the old `Sensei_Notices` because it should be used regardless filters or anything like that, and we can attach to filters, blocks, and so on, depending on the scenario we want (and in the scope that is using that). Also, it contains a "context" feature, so we can handle notices for different contexts at the same time. In the future, we can refactor the `Sensei_Notices` to use that.
* It also adds a new block to the Course Theme to show the main notices on the pages, and it uses the `Sensei_Context_Notices`.

### Testing instructions


* Enable the feature flag: `add_filter( 'sensei_feature_flag_course_theme', '__return_true' );`
* Create a course with 3 lessons with quizzes.
* Set a pass required for the quizzes.
* As a student, submit and keep one quiz ungraded.
* Submit another quiz with enough points to pass, and grade it.
* Submit another quiz without enough points to pass, and grade it.
* Select the "Sensei Theme" as the "Course Theme" (sidebar).
* Go to the 3 lessons in the frontend, and make sure you see the 3 different scenarios with their respective notices (see screenshots).


### New/Updated Hooks

* `sensei_context_notices`: Filter notices when using the Context Notices class.


### Screenshot / Video

The default notice. Only the main text ("Notice text here") is required.
<img width="1326" alt="Screen Shot 2021-11-26 at 15 08 27" src="https://user-images.githubusercontent.com/876340/143624867-4aea0c1c-8729-42c5-ab61-3e8a30dd07f9.png">

Awaiting grade status:
<img width="1317" alt="Screen Shot 2021-11-26 at 15 30 15" src="https://user-images.githubusercontent.com/876340/143624872-74eb3fb2-7386-4ad6-9cd9-a5e3aac5973c.png">

Graded and approved:
<img width="1307" alt="Screen Shot 2021-11-26 at 15 27 46" src="https://user-images.githubusercontent.com/876340/143624871-4501bfab-0678-4fb3-97d9-0daacdbb6ac7.png">

Graded and not approved:
<img width="1316" alt="Screen Shot 2021-11-26 at 15 22 34" src="https://user-images.githubusercontent.com/876340/143624869-93a19029-10a0-49f9-adcd-bde48c537643.png">
